app/services/foreman/provision/ssh.rb

```diff
@@ -72,39 +72,24 @@ def logger
   end
 
   def initiate_connection!
-    Timeout.timeout(360) do
-      begin
-        Timeout.timeout(8) do
-          ssh.run('pwd')
+    begin
```

Comment:
I am not super familar with ruby, but I think this may be a false positive as I need the nested begin/rescue's to handle to two nexted Timeouts. But if I am incorrect let me know and I will gladly fix it.